astropy/convolution/tests/test_kernel_class.py

```diff
@@ -473,15 +473,16 @@ def test_kernel_normalization_mode(self):
             kernel = CustomKernel(np.ones(3))
             kernel.normalize(mode='invalid')
 
-    def test_kernel_normalization_large(self, recwarn):
+    def test_kernel_normalization_large(self):
         """
         Test that a warning is issued when the inverse normalization factor
         is large.
         """
-        kernel = CustomKernel(np.ones(3) * 1.e-3)
-        kernel.normalize()
-        w = recwarn.pop(AstropyUserWarning)
-        assert issubclass(w.category, AstropyWarning)
+        with pytest.warns(AstropyUserWarning) as w:
```

Comment:
I won't argue with code that works, though since everywhere else we seem to use `catch_warnings`, maybe that would be best here too? (Or possibly, we should deprecate that if `pytest` can handle it anyway -- but that would be for another PR, obviously.)